contracts-cli/contract.js

```diff
@@ -0,0 +1,128 @@
+const { prompt } = require('inquirer')
+const { utils } = require('web3')
+
+const config = require('./config')
+const customHandlers = require('./customHandlers')
+const { log } = require('./cli-utils')
+const { GAS_LIMIT, GAS_PRICE } = require('./constants')
+
+const callCustomHandler = (contractName, method, web3Contract, ethNetwork, account) => {
+  const customHandler = customHandlers[contractName]
+  if (customHandler && customHandler[method]) {
+    customHandler[method](web3Contract, ethNetwork, account)
+    return true
+  }
+}
+
+const getWeb3Contract = async (name, abi, web3, ethNetwork) => {
+  var contractAddress
+  if (config.contractAddresses && config.contractAddresses[name]) {
+    contractAddress = config.contractAddresses[name][ethNetwork]
+  } else {
+    const answers = await prompt({
+      type: 'input',
+      name: 'address',
+      message: 'Enter contract address',
+    })
+    contractAddress = answers.address
+  }
+  return new web3.eth.Contract(abi, contractAddress)
+}
+
+const callMethod = async (
+  web3Contract,
+  contractName,
+  method,
+  account,
+  ethNetwork
+) => {
+  if (callCustomHandler(
+    contractName,
+    method,
+    web3Contract,
+    ethNetwork,
+    account)
+  ) {
+    return
+  }
+  const abiMethod = web3Contract._jsonInterface.find(m => m.name == method)
+  const questions = abiMethod.inputs.map(m => {
+    const q = {
+      type: 'input',
+      name: m.name,
+      message: `Enter ${m.name} (${m.type}): `,
+    }
+    // If uint, ask if it needs unit conversion
+    if (m.type.slice(0, 4) === 'uint') {
+      q.extraQuestions = [
+        {
+          type: 'confirm',
+          name: 'confirm',
+          message: `Do you need to convert unit? `,
+        },
+        {
+          type: 'input',
+          name: 'decimals',
+          message: `Enter conversion unit (e.g. gwei, finney): `,
+        },
+        {
+          type: 'list',
+          name: 'conversion',
+          message: `Conversaion type: `,
+          choices: ['fromWei', 'toWei'],
+        }
+      ]
+    }
+    return q
+  })
+  const answers = await prompt(questions)
+  const args = []
+  for (i = 0; i < questions.length; i++) {
+    const q = questions[i]
+    var arg = answers[q.name]
```

Comment:
Not an issue but it's usually recommended to use `let` instead of `var` for variables that are expected to be reassigned, and `const` when they shouldn't. JS used to only have `var` so it does the same as `let`, but `let` makes it more explicit in newer codebases that this is "not `const`".